2 add new MDX guides for "Configuring basemaps" and "Viewing the program" (#4)
parkerziegler/cartokit-docs: src/content/docs/guides/layer-types.mdx

Reviewer: ```suggestion
![A Polygon layer in cartokit displaying urban areas in the United States.](/polygon.png)
```

`cartokit` currently supports six layer types. The available layer types for a given layer are dependent on the `geometry` of the input data.

#### Supported Input Geometries

- `Point`
- `MultiPoint`
- `Polygon` (uses centroid)
- `MultiPolygon` (uses centroid)
- `LineString` (uses centroid)
- `MultiLineString` (uses centroid)

> 

Reviewer: We should actually have this section duplicated as a subsection for each layer type, because the supported input geometry types _differ_ based on the layer type (it's the weird distinction between the _data_ type and the layer type). For example, the **Point** section below should read as follows:

```md
### Point

The **Point** layer type displays all features of a layer as points with a consistent, configurable radius. This type is the default type for `Point` and `MultiPoint` GeoJSON datasets.

#### Supported Input Geometries

- `Point`
- `MultiPoint`
- `Polygon` (uses centroid)
- `MultiPolygon` (uses centroid)
- `LineString` (uses centroid)
- `MultiLineString` (uses centroid)
```

whereas the **Polygon** section farther down should read as follows:

```md
### Polygon

The **Polygon** layer type displays geographic areas as filled shapes with configurable fill color and border style. This type is the default type for `Polygon` and `MultiPolygon` GeoJSON datasets.

#### Supported Input Geometries

- `Polygon`
- `MultiPolygon`
```

You _can't_ magically get `Polygon` data from `Point` data, but you can go the other way using centroids. We want to capture those kinds of constraints for each layer type! Let me know if you're unsure about any of them!

### Proportional Symbol

> The **Proportional Symbol** layer type allows scaling of point symbols based on the value of a selected attribute for each data point. This is one option type for `MultiPoint`, `MultiPolygon` and `MultiLineString` GeoJSON datasets.

Reviewer: Thanks so much for catching this! Let's just replace this second sentence with the `#### Supported Input Geometries` section for this layer type.

![A Proportional Symbol layer in cartokit consisting of all natural gas-powered electricity plants in the United States and their capacities.](/proportional-symbol.png)

### Heatmap

> The **Heatmap** layer type displays the density of data using a color gradient with intensity styled by a **Constant** concentration of points or weighted **Range** of a selected attribute value. This is one option type for `MultiPoint` GeoJSON datasets.

Reviewer: A good thing to note is that `Point` and `MultiPoint` (and `LineString` and `MultiLineString`, `Polygon` and `MultiPolygon`) are all subtly different. A GeoJSON `FeatureCollection` consisting of `Point` features means that each feature in the dataset has _exactly_ one associated geographic point. But a `MultiPoint` feature may actually associate one feature with _several_ geographic points.

A good example to think about is something like islands (e.g., Hawaii). Hawaii typically gets represented in GeoJSON datasets as a `MultiPolygon` feature. It's conceptually one "entity" (the state of Hawaii) but its `geometry` consists of multiple `Polygon`s (one for each island).

In general, we want to indicate that we support both, since some tools / algorithms have restrictions that they don't support multi-part geometries (and it's nice that `cartokit` does)!

### Choropleth

> The **Choropleth** layer type colors polygons based on the value of a selected attribute, either **Categories** or a numerical **Range** with a configurable scale. This is one option type for `MultiPolygon` GeoJSON datasets.

Reviewer: ```suggestion
The **Choropleth** layer type colors polygons based on the value of a selected attribute, using either **Categories** or a numerical **Range** with a configurable scale.

#### Supported Input Geometries

- `Polygon`
- `MultiPolygon`
```

> ![A Choropleth Symbol layer in cartokit visualizing the predicted change in deaths linked to temperature by 2080 across the world.](/choropleth.png)

Reviewer: ```suggestion
![A Choropleth layer in cartokit visualizing the predicted change in deaths linked to temperature by 2080 across the world.](/choropleth.png)
```

> ![A Dot Density Symbol layer in cartokit visualizing the predicted change in deaths linked to temperature by 2080 across the world.](/dot-density.png)

Reviewer: ```suggestion
![A Dot Density layer in cartokit visualizing the predicted change in deaths linked to temperature by 2080 across the world.](/dot-density.png)
```
